spec/enumerable_spec.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+# spec/calculator_spec.rb
+# require File.expand_path('./enumerable.rb')
+require File.expand_path('../enumerable.rb')
+
+RSpec.describe Enumerable do
+  #=begin
+  describe '#my_each_with_index' do
+    let(:arr) { [2, 4, 6] }
+    let(:result) { [] }
+    it 'find each item with his index' do
+      arr.my_each_with_index { |x, i| result.push("#{i} index of #{x}") }
+      expect(result).to eql(['0 index of 2', '1 index of 4', '2 index of 6'])
+    end
+    it 'return the original array' do
+      expect(arr.my_each_with_index do |x, i|
+               result.push("#{i} index \
+                     of #{x}")
+             end).to eql(arr)
+    end
+  end
+  #=end
+
+  #=begin
+  describe '#my_select' do
+    let(:arr) { [2, 4, 6, 8, 10, 12] }
```

Comment:
This repeats several times in your tests... you can simply add it after line 7 and make it available to all subsequent tests...

You can do this for all let variable declarations